### DISCUSSION TITLE
How can i load sqlite extension (.a,.so,.dylib) file.

### DISCUSSION THREAD

**Discussion by drunisa007** [simolus3/drift#2526] [Q&A]
in rust, rusqlite crate have `load_extension` function to load sqlite extension.
```  db.conn
            .load_extension(load_extension_path, None)
            .unwrap();
```
 how can I achieve this in drift package.

**Comment by simolus3** [comment]
You can load extensions with the [sqlite3](https://pub.dev/documentation/sqlite3/latest/sqlite3/Sqlite3/ensureExtensionLoaded.html) package, [this documentation page](https://pub.dev/documentation/sqlite3/latest/sqlite3/SqliteExtension-class.html) links some examples.

The sqlite3 builds you get from `sqlite3_flutter_libs` don't include `sqlite3_load_extension` due to security concerns (and since consistently loading a .so is not that easy across platforms), but you can manually open the dynamic library in Dart and then load the extension through that: https://github.com/simolus3/sqlite3.dart/blob/a9a379494c6b8d58a3c31cf04fe16e83b49130f1/sqlite3/test/ffi/sqlite3_test.dart#L84-L86 If you do that before using drift, the extension will be available to drift databases.
